src/main/java/com/intuit/graphql/orchestrator/schema/fold/UnifiedXtextGraphFolder.java

```diff
@@ -148,10 +161,42 @@ private void resolveTypeConflicts(
             conflictingType,
             existingType,
             currentGraph.getServiceProvider().isFederationProvider());
+        mergeSharedType(existingType, conflictingType, accumulator, currentGraph);
       }
     }
   }
 
+  private void mergeSharedType(TypeDefinition existingType, TypeDefinition incomingType,
+      UnifiedXtextGraph accumulator, XtextGraph incomingXtextGraph) {
+    boolean isFederated = incomingXtextGraph.getServiceProvider().isFederationProvider();
+    if (isFederated && isEntity(existingType) && isEntity(incomingType)) {
+      mergeEntityTypes(existingType, incomingType, accumulator, incomingXtextGraph);
+    }
+  }
+
+  private void mergeEntityTypes(TypeDefinition existingType, TypeDefinition incomingType,
+      UnifiedXtextGraph accumulator, XtextGraph incomingXtextGraph) {
+    if(isBaseType(existingType) && isEntityExtensionType(incomingType)) {
+      // TODO consider calling EntityTypeMerging here
+      incomingXtextGraph.getTypes().put(existingType.getName(), existingType);
+      updateTypeReferencesInAnXtextGraph(incomingType, existingType, incomingXtextGraph);
```

Comment:
lets comment this line like we discussed and add an explanation for if...else block

Reply:
done